src/NHibernate/Cfg/Configuration.cs

```diff
@@ -275,23 +273,20 @@ public bool HasNonIdentifierPropertyNamedId(string className)
 			public Dialect.Dialect Dialect { get; }
 		}
 
-		private IMapping mapping;
-
 		protected Configuration(SettingsFactory settingsFactory)
 		{
-			InitBlock();
 			this.settingsFactory = settingsFactory;
 			Reset();
 		}
 
-		private void InitBlock()
+		public virtual IMapping BuildMapping()
 		{
-			mapping = BuildMapping();
+			return new Mapping(this);
 		}
 
-		public virtual IMapping BuildMapping()
+		private IMapping BuildMapping(Dialect.Dialect dialect)
```

Comment:
I don't think I want to make this method public.

Reply:
To me, it looks like it can be overridden to adjust session factory creation logic. So making it private and suggesting the use of `BuildSessionFactory` instead is not exactly an equivalent. But really, I have no knowledge of how exactly overriding it can help someone and if it's really useful.

Reply:
> To me, it looks like it can be overridden to adjust session factory creation logic.

The `IMapping` interface is passed to the session factory, but really not used there (only some piece of code - I did not look carefully - about temporary tables) and, probably, can be removed. So it does not really help to session factory logic. 

> suggesting the use of `BuildSessionFactory` instead is not exactly an equivalent.

Yes, it is not an equivalent. `IMapping` produced by `BuildMapping` is a half-baked mapping at some intermediate step, while `SessionFactory` is fully resolved working instance of `IMapping`. So, yes, it is not equivalent, but more correct way to get `IMapping` unless you're doing some black-magic type of configurations and introspections early on.

Reply:
nit: I would keep it public

Reply:
Made public virtual